lib/Migration/Version150003Date20201004181908.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\News\Migration;
+
+use Closure;
+use Doctrine\DBAL\Types\Type;
+use OC\DB\QueryBuilder\QueryBuilder;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
+ */
+class Version150003Date20201004181908 extends SimpleMigrationStep {
+
+    protected $queryBuilder;
+
+    public function __construct(QueryBuilder $queryBuilder)
+    {
+        $this->queryBuilder = $queryBuilder;
+    }
+
+    /**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options) {
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options) {
+        /** @var ISchemaWrapper $schema */
+        $schema = $schemaClosure();
+
+        if ($schema->hasTable('news_items') &&
+            !$schema->getTable('news_items')->hasForeignKey('feed')) {
+            $schema->getTable('news_items')
+                ->getColumn('feed_id')
+                ->setUnsigned(true);
+
+            $schema->getTable('news_items')
+                   ->addForeignKeyConstraint(
+                       $schema->getTable('news_feeds')->getName(),
+                       ['feed_id'],
+                       ['id'],
+                       ['onDelete' => 'CASCADE'],
+                       'feed'
+                   );
+        }
+
+        if ($schema->hasTable('news_feeds') &&
+            !$schema->getTable('news_feeds')->hasForeignKey('folder')) {
+            $schema->getTable('news_feeds')
+                   ->getColumn('folder_id')
+                   ->setUnsigned(true)
+                   ->setNotnull(false)
+                   ->setDefault(null);
+
+            $schema->getTable('news_feeds')
+                   ->addForeignKeyConstraint(
+                       $schema->getTable('news_folders')->getName(),
+                       ['folder_id'],
+                       ['id'],
+                       ['onDelete' => 'CASCADE'],
+                       'folder'
+                   );
+        }
+
+        return $schema;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options) {
+	    $this->queryBuilder->update('news_feeds')
+                           ->set('folder_id', null)
+                           ->where('folder_id = 0');
```

Comment:
Without having it tested .. are you sure this is going to work here, after adding the constraints, since the `folder_id` typically has to be valid for the successfull creation of the foreign keys.

For MySQL there is an option to disable the check, which could be executed as a preSchemaChange, but I don't know how the other DB types handle this: `SET FOREIGN_KEY_CHECKS=0;` (1 for re-enabling the check)

Reply:
I'm not sure this will work, but I'm not able to test this completely right now. I was hoping someone would be able to tell me once I got this to actually build

Reply:
Alright, I'll test it as soon as I have enough time .. the CI seems to be successful by now. 👍

Reply:
Okay, last push actually worked. Would you be able to check if it worked for you?

Reply:
Missed the comment, thanks in advance.